pkg/auth/auth.go

```diff
@@ -153,3 +161,182 @@ func (c *Client) RevokeToken(ctx context.Context, req *RevokeTokenRequest) error
 	_, err := c.post(ctx, "/token-revoke", req)
 	return err
 }
+
+type Tokenstore struct {
+	AccessToken             string    `json:"access_token"`
+	AccessTokenGeneratedAt  time.Time `json:"access_token_generated_at"`
+	RefreshToken            string    `json:"refresh_token"`
+	RefreshTokenGeneratedAt time.Time `json:"refresh_token_generated_at"`
+}
+
+// Helper struct holds the username, password and token data
+type Helper struct {
+	Username  string
+	Password  string
+	TokenData *Tokenstore
+	apiMutex  sync.Mutex
+	API       API
+
+	fileMutex sync.Mutex
+}
+
+// NewHelper creates a new Helper instance
+func NewHelper(api API) (*Helper, error) {
+	err := godotenv.Load()
+	if err != nil {
+		return nil, fmt.Errorf("error loading .env file: %v", err)
+	}
+
+	username := os.Getenv("WME_USERNAME")
+	password := os.Getenv("WME_PASSWORD")
+	if username == "" || password == "" {
+		return nil, errors.New("username or password not set in .env file")
+	}
+
+	return &Helper{
+		Username: username,
+		Password: password,
+		API:      api,
+	}, nil
+}
+
+// GetAccessToken manages the token state and returns a valid access token
+func (h *Helper) GetAccessToken() (string, error) {
+	h.apiMutex.Lock()
+	defer h.apiMutex.Unlock()
```

Comment:
I guess we can remove the top level lock now, as we are doing lock on file write at the lower level.

Reply:
Removed